src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClientFactory.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.Azure.WebJobs.Extensions.DurableTask.Options;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask.ContextImplementations
+{
+    /// <summary>
+    ///     Factory class to create Durable Client to start works outside an azure function context.
+    /// </summary>
+    public class DurableClientFactory : IDurableClientFactory
+    {
+        private readonly DurableTaskExtension durableTaskConfig;
+
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="DurableClientFactory"/> class.
+        /// </summary>
+        /// <param name="durableTaskConfig">Configuration for the Durable Functions extension.</param>
+        public DurableClientFactory(DurableTaskExtension durableTaskConfig)
+        {
+            this.durableTaskConfig = durableTaskConfig;
+        }
+
+        /// <summary>
+        /// Gets a <see cref="IDurableClient"/> using configuration from a <see cref="DurableClientOptions"/> instance.
+        /// </summary>
+        /// <param name="durableClientOptions">options containing the client configuration parameters.</param>
+        /// <returns>Returns a <see cref="IDurableClient"/> instance. The returned instance may be a cached instance.</returns>
+        public IDurableClient CreateClient(DurableClientOptions durableClientOptions)
```

Comment:
It also could be CreateExternalClient and remove the Boolean property and set always true in the attribute to avoid this overhead, I thought it also can be used inside an azure function somewhere